bundles/org.openhab.binding.iammeter/src/main/java/org/openhab/binding/iammeter/internal/Iammeter3080THandler.java

```diff
@@ -0,0 +1,153 @@
+ * sent to one of the channels.
+ *
+ * @author Yang Bo - Initial contribution
+ */
+
+@NonNullByDefault
+public class Iammeter3080THandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(Iammeter3080THandler.class);
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private IammeterConfiguration config;
+    private static final int TIMEOUT_MS = 5000;
+    private final ExpiringCache<Boolean> refreshCache = new ExpiringCache<>(Duration.ofSeconds(5), this::refresh);
+
+    public Iammeter3080THandler(Thing thing) {
+        super(thing);
+        config = getConfiguration();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            refreshCache.getValue();
+        }
+    }
+
+    @Override
+    public void initialize() {
+        ScheduledFuture<?> refreshJob = this.refreshJob;
+        config = getConfiguration();
+        if (refreshJob == null) {
+            refreshJob = scheduler.scheduleWithFixedDelay(this::refresh, 0, config.refreshInterval, TimeUnit.SECONDS);
+            this.refreshJob = refreshJob;
+            updateStatus(ThingStatus.UNKNOWN);
+        }
+    }
+
+    @SuppressWarnings("null")
+    private boolean refresh() {
+        refreshCache.invalidateValue();
+        IammeterConfiguration config = this.config;
+        try {
+            String httpMethod = "GET";
+            String url = "http://" + config.username + ":" + config.password + "@" + config.host + ":" + config.port
+                    + "/monitorjson";
+            String content = "";
+            InputStream stream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8));
+
+            String response = HttpUtil.executeUrl(httpMethod, url, stream, null, TIMEOUT_MS);
+            JsonElement iammeterDataElement = new JsonParser().parse(response);
+            JsonObject iammeterData = iammeterDataElement.getAsJsonObject();
+            String keyWord = "Datas";
+            if (iammeterData.has("Datas") && iammeterData.has("SN")) {
+                String groups[] = { "powerPhaseA", "powerPhaseB", "powerPhaseC" };
+                for (int row = 0; row < groups.length; row++) {
+                    String gpName = groups[row];
+                    List<Channel> chnList = getThing().getChannelsOfGroup(gpName);
+                    for (IammeterWEM3080Channel channelConfig : IammeterWEM3080Channel.values()) {
+                        Channel chnl = chnList.get(channelConfig.ordinal());
+                        if (chnl != null) {
+                            State state = getDecimal(iammeterData.get(keyWord).getAsJsonArray().get(row)
+                                    .getAsJsonArray().get(channelConfig.ordinal()).toString(), channelConfig.getUnit());
+                            updateState(chnl.getUID(), state);
+                        }
+                    }
```

Comment:
Related to the class splitting. I didn't intend for you to completely copy paste the code as it duplicates a lot of code. Better make an `abstract` base class with all the logic and let it call methods to do the specific code. The code I selected here would be in the sub class. The method would get the response as parameter. You can make the method `getDecimal`  `protected` (better rename it `getQuantityState` probably`.

Reply:
abstract class made.

Reply:
abstract class made.